modules/claims_api/app/controllers/claims_api/v2/application_controller.rb

```diff
@@ -191,6 +191,20 @@ def build_target_veteran(veteran_id:, loa:) # rubocop:disable Metrics/MethodLeng
         target_veteran[:va_profile] = ClaimsApi::Veteran.build_profile(mpi_profile.birth_date)
         target_veteran
       end
+
+      def file_number_check
+        if !target_veteran.mpi.birls_id.empty?
+          @file_number = target_veteran.mpi.birls_id
+        elsif !auth_headers['va_eauth_birlsfilenumber'].empty?
+          @file_number = auth_headers['va_eauth_birlsfilenumber'].present?
+        else
+          ClaimsApi::Logger.log('missing_file_number', detail: "missing_file_number on request: #{request}")
+
+          raise ::CommonExceptions::UnprocessableEntity.new(detail:
+            "Unable to locate Veteran's 'File Number' in Master Person Index (MPI). " \
+            'Please submit an issue at ask.va.gov or call 1-800-MyVA411 (800-698-2411) for assistance.')
+        end
```

Comment:
@FonzMP questions: 
1. do we want to check both the auth headers & the mpi.birls_id, or just mpi?
2. how should we log? is it ok to log the request since this is in the app controller, if not, what else can we log?

Reply:
1. I don't know if I really like the idea of falling back to the birls in the auth headers. It's dependent on the EVSS Headers which is logic that could be removed relatively soon? It's also just coming from the `target_veteran.birls_id` if you look at the EVSS Headers logic. https://github.com/department-of-veterans-affairs/vets-api/blob/de24868880b901b51677745fbaf363f9f421072f/lib/evss/auth_headers.rb#L27 . We could just run off of `target_veteran.birls_id` unless the `target_veteran.mpi.birls_id` is different?

2. Hmmm.... I think we're allowed to log ICN's as they're on an internal system, but I don't think they want that any more. At this point in the code I'm not sure if we would be able to log much that would be acceptable. Let me talk with Drew.